js/btcex.js

```diff
@@ -2075,6 +2077,42 @@ module.exports = class btcex extends Exchange {
         return result;
     }
 
+    async setMarginMode (marginMode, symbol = undefined, params = {}) {
+        /**
+         * @method
+         * @name btcex#setMarginMode
+         * @description set margin mode to 'cross' or 'isolated'
+         * @see https://docs.btcex.com/#modify-perpetual-instrument-margin-type
+         * @param {string} marginMode 'cross' or 'isolated'
+         * @param {string|undefined} symbol unified market symbol
+         * @param {object} params extra parameters specific to the btcex api endpoint
+         * @returns {object} response from the exchange
+         */
+        await this.signIn ();
+        await this.loadMarkets ();
+        this.checkRequiredSymbol ('setMarginMode', symbol);
+        const market = this.market (symbol);
+        if (market['type'] !== 'swap') {
+            throw new BadRequest (this.id + ' setMarginMode() supports swap contracts only');
+        }
+        marginMode = (marginMode === 'isolated') ? 'isolate' : 'cross';
```

Comment:
@Dan-krm  Shall we throw error if margin mode is not either `isolated` or `cross`? We did that in binance:

```JS
        if ((marginMode !== 'ISOLATED') && (marginMode !== 'CROSSED')) {
            throw new BadRequest (this.id + ' marginMode must be either isolated or cross');
        }
```